controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -174,6 +183,71 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) reconcileOnboardingSecret() error {
+	secret := &corev1.Secret{}
+	secret.Name = fmt.Sprintf("onboarding-token-%d", util.FnvHash(string(r.storageConsumer.UID)))
+	secret.Namespace = r.storageConsumer.Namespace
+	if err := r.get(secret); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get secret %s: %v", secret.Name, err)
+	}
+
+	if !r.storageConsumer.Spec.Enable {
+		// if not enabled, ensure secret presence
+		if secret.UID == "" {
+			if err := r.own(secret); err != nil {
+				return err
+			}
+			storageCluster, err := util.GetStorageClusterInNamespace(r.ctx, r.Client, r.namespace)
+			if err != nil {
+				return err
+			}
+			salt := uuid.New().String()
+			token, err := util.GenerateClientOnboardingToken(
+				tokenLifetimeInHours,
+				onboardingPrivateKeyFilePath,
+				nil,
+				storageCluster.UID,
+				salt,
+			)
+			if err != nil {
+				return err
+			}
+			secret.Data = map[string][]byte{
+				"ticket": []byte(token),
+			}
+			util.AddAnnotation(secret, "token-expiry-utc", time.Now().Add(tokenLifetimeInHours*time.Hour).UTC().String())
+			// as salt is random we are annotating it on secret as well to read later if we fail to annotate consumer in current reconcile
+			util.AddAnnotation(secret, TicketAnnotation, salt)
```

Comment:
Bad strategy, if the intent of salt is to be an identifier and not a just random junk, you need to be able to consistently reproduce it.  Please reuse something that already exists or use a conversion of something that already exists. 


Suggestion: A fixed length hash of the (clusterd + storage cluster name)

Reply:
using storageconsumer name directly as the identifier.